map/place_page_info.cpp

```diff
@@ -122,7 +122,8 @@ std::string Info::FormatSubtitle(bool withMainType) const
     append(m_bookmarkCategoryName);
 
   // Types
-  append(GetLocalizedAllTypes(withMainType));
+  if (withTypes)
+      append(GetLocalizedAllTypes(withMainType));
```

Comment:
then probably most of funcs from below like `GetLocalizedCuisines()` shouldn't be called too?

as they apply to features only (not "empty space" bookmarks or "my position")

Reply:
also metadata could possibly exist on features only, right?

Reply:
Right. I wanted to keep the changes in this fix minimal, but you are right that probably most of them don't make sense to be executing when it's not a feature

Reply:
Sure! It looks good as is.

Could be improved in a separate PR if you feel like it :pray:

Reply:
If it's easy to fix, let's do it right now. There's a rule: if you can do something in a few minutes now, do not postpone it, later it will take more time )

Reply:
Actually I'm not sure about bookmarks as they save/contain feature types too, e.g.
```
      <mwm:featureTypes>
        <mwm:value>tourism-attraction</mwm:value>
        <mwm:value>natural-cave_entrance</mwm:value>
        <mwm:value>historic-ruins</mwm:value>
      </mwm:featureTypes>
```

IDK how they're used.

Maybe they're displayed instead of underlying feature's types for cases when a feature had been changed or is long gone since bookmarking..

Reply:
> If it's easy to fix, let's do it right now. There's a rule: if you can do something in a few minutes now, do not postpone it, later it will take more time )

It could be easy or we could break something else. Since it’s not clear how this is going to turn out I would prefer to keep the scope of this fix reduced.

Reply:
Am I correctly understanding that introduced changes that are fixed now are still not available in production? Then the "reducing the scope" rule doesn't apply here. It's all new code that can be/should be properly fixed.

Reply:
This PR is fixing new code that was added in last PR. @pastk was proposing to modify existing FormatSubtitle code that has been in production for a long time

Reply:
I agree it'd be better to do a refactoring after the upcoming release.
This PR fixes an issue with already merged changes.

Reply:
Looks like the discussion is already taking more time in summary than the proper fix )

Did you check the code? Is it safe to work when types are empty? Did you test it?

Is adding an if (or early return) too risky?

Reply:
I've added an early return. Following code is not needed if withTypes is false